source/extensions/common/redis/redirection_mgr_impl.cc

```diff
@@ -39,14 +51,36 @@ bool RedirectionManagerImpl::onRedirection(const std::string& cluster_name) {
     uint64_t last_callback_time_ms = info->last_callback_time_ms_.load();
     if (!last_callback_time_ms ||
         (now >= (last_callback_time_ms + info->min_time_between_triggering_.count()))) {
-      info->redirects_count_++; // ignore redirects during min time between triggering
-      if ((info->redirects_count_ >= info->redirects_threshold_) &&
+      std::atomic<uint32_t>* count;
+      uint32_t threshold;
+      switch (event_type) {
+      case EventType::Redirection: {
+        count = &(info->redirects_count_);
+        threshold = info->redirects_threshold_;
+        break;
+      }
+      case EventType::DegradedHost: {
+        count = &(info->host_degraded_count_);
+        threshold = info->host_degraded_threshold_;
+        break;
+      }
+      case EventType::Failure: {
+        count = &(info->failures_count_);
+        threshold = info->failure_threshold_;
+        break;
+      }
+      }
+      if (threshold <= 0) {
+        return false;
+      }
+      // ignore redirects during min time between triggering
+      if ((++(*count) >= threshold) &&
           (info->last_callback_time_ms_.compare_exchange_strong(last_callback_time_ms, now))) {
         // last_callback_time_ms_ successfully updated without any changes since it was
         // initially read. This thread is allowed to post a call to the registered callback
         // on the main thread. Otherwise, the thread would be ignored to prevent over-triggering
         // cluster callbacks.
-        info->redirects_count_ = 0;
+        *count = 0;
```

Comment:
Is it necessary to zero this?

Reply:
yes, this is critical.  Otherwise, we'll keep refreshing.